leafmap/ai.py

```
@@ -0,0 +1,774 @@
+    "importlib",
+    "ctypes",
+    "pickle",
+    "shelve",
+}
+
+
+class SafetyError(Exception):
+    """安全校验未通过。"""
+
+
+def _check_ast_safety(code: str) -> None:
```

Ruff (0.16.1)

[warning] Too many branches (13 > 12)

(PLR0912)